### PR TITLE
add first chapter refactor of the tutorial

### PR DESCRIPTION
**Pull request by pablogh-2000**

### CODE REVIEW

File: tutorials/chapter1.md [hunk 1]
```diff
@@ -0,0 +1,650 @@
...
+* Although not required beforehand, there will be use of [YAML language](https://yaml.org/).
+* Klepsydra recommends the use of the [composition root](https://medium.com/@cfryerdev/dependency-injection-composition-root-418a1bb19130) and [dependency injection](https://en.wikipedia.org/wiki/Dependency_injection) patterns. These two concepts appear repeatedly throughout this tutorial.
+
+For the ROS tutorial, basic knowledge of ROS is required and similarly for the DDS tutorial, RTI DDS understanding is required.
+
+## Technical requirements
+
+Installation of Klepsydra Core, Optionally ROS, DDS.
+
+Klepsydra is installed (by default) to /usr/local by CMake. To develop an application using Klepsydra use the cmake command `find_package(Klepsydra)` in the CMakeLists file.. With this command, the Klepsydra includes are available in the variable `KLEPSYDRA_INCLUDE_DIRS` and the libraries are available using the variable `KLEPSYDRA_CORE_LIBRARIES`. The CMakeLists file provided for compiling the examples in the tutorial demonstrate how these two variables are used.
```
Comment (guillermo-sarabia) [inline tutorials/chapter1.md]: Hyper link to the CMakelists?

File: tutorials/chapter1.md [hunk 2]
```diff
@@ -0,0 +1,650 @@
...
+
+## Chapter 1
+
+
+### Introduction
+
+Klepsydra API is split into main groups:
+
+* **Application API**. A simple Publisher / Subscriber API, this API is used by the application or customer code to build the functionality.
+* **Composition API**. This API is used to create and assembly publisher subscriber instances. Examples are: EventLoop, ROS Connector, etc.
```
Comment (guillermo-sarabia) [inline tutorials/chapter1.md]: How does this affect the structure of the tutorial? I guess **Application API** is somehow simpler than **Composition API** Are we going to cover both?

File: tutorials/chapter1.md [hunk 3]
```diff
@@ -0,0 +1,650 @@
...
+
+This tutorial does not intend to cover the complete API of Klepsydra Core, but to explain the key programming features of it. For a full comprehensive description of the API, please refer to the api-doc folder of the kpsr-core repository. @TODO: add link
+
+### Example 1: The "Hello world!" example.
+
+Lets start with a simple example. First, let's create a class that publishes a string:
```
Comment (guillermo-sarabia) [inline tutorials/chapter1.md]: Hyperlink to file?

File: tutorials/chapter1.md [hunk 4]
```diff
@@ -0,0 +1,650 @@
...
+
+   eventloop.getSubscriber<std::string>()->removeListener("example1");
+
+   eventloop.stop();
+}
+```
+
+This version of the main is slighlty different from the previous one in the following elements:
+
+* The instatiation of the ```EventLoopMiddlewareProvider``` does not take any template, nor the ```std::function``` pair for the smart object pool. The reason for that is that the ```EventLoop``` is a general memory sharing system that can handle data of any type, it only requires the size of the ring buffer as template parameter. It is only when invoking ```getPublisher``` and ```getSubscriber``` that types need to be specified. The ```EventEmitter``` can only handle one publisher / subsriber pair, this is why the templating and smart pool configuration happen at the declaration and construction level.
```
Comment (guillermo-sarabia) [inline tutorials/chapter1.md]:
```suggestion
* The instatiation of the ```EventLoopMiddlewareProvider``` does not take any template, nor the ```std::function``` pair for the smart object pool. The reason for that is that the ```EventLoop``` is a general memory sharing system that can handle data of any type, it only requires the size of the ring buffer as template parameter. It is only when invoking ```getPublisher``` and ```getSubscriber``` that types need to be specified. The ```EventEmitter``` can only handle one publisher / subscriber pair, this is why the templating and smart pool configuration happen at the declaration and construction level.
```

File: tutorials/chapter1.md (hunk 1 above)
Comment (pablogh-2000) [reply tutorials/chapter1.md]: I have restructured the tutorial doc a bit now.

File: tutorials/chapter1.md (hunk 2 above)
Comment (pablogh-2000) [reply tutorials/chapter1.md]: Both are fully covered in chapter 1. Let me know if this is not clear?

File: tutorials/chapter1.md (hunk 3 above)
Comment (pablogh-2000) [reply tutorials/chapter1.md]: I added some...

File: tutorials/chapter1.md (hunk 4 above)
Comment (pablogh-2000) [reply tutorials/chapter1.md]: Fixed thanks

**Pull request merged**